contracts/src/interfaces/IGateway.sol

```diff
@@ -78,6 +78,12 @@ interface IGateway {
     /// @dev Emitted when a command is sent to register a new wrapped token on AssetHub
     event TokenRegistrationSent(address token);
 
+    /// @dev the address of the BEEFY light client used to verify messages.
+    function BEEFY_CLIENT() external returns (address);
+
+    // @dev the address of the agent executor implementaiton.
+    function AGENT_EXECUTOR() external returns (address);
+
```

Comment:
Unnecessary?

Reply:
Our bindings are generated off IGateway for the relayer and the contract-types npm package. So this is helpful for that.

Reply:
The problem is that I want `IGateway` to be more for user facing APIs, not infrastructure concerns. In fact, I want to remove more non-user APIs from IGateway.

Can't you generate bindings for `Gateway.sol`, and then use `IGateway.implemention()` to get the implementation contract?